File: THCS/2022_2023/VungTau_2223/Bai2/GIFT.py
def main():
    # Đọc dữ liệu từ file GIFT.INP
    with open("GIFT.INP", "r") as infile:
        N, K = map(int, infile.readline().strip().split())
        gifts = [int(infile.readline().strip()) for _ in range(N)]

    # Sắp xếp danh sách phần quà theo giá trị tăng dần
    gifts.sort()

    count = 0  # Số lượng phần quà đã chọn
    lastGift = -1  # Giá trị của phần quà cuối cùng đã chọn

    for gift in gifts:
        # Kiểm tra xem phần quà có thể được chọn hay không
        if lastGift == -1 or gift >= lastGift + K:
            count += 1
            lastGift = gift  # Cập nhật giá trị của phần quà đã chọn

    # Ghi kết quả vào file GIFT.OUT
    with open("GIFT.OUT", "w") as outfile:
        outfile.write(f"{count}\n")  # Ghi số lượng phần quà đã chọn

if __name__ == "__main__":
    main()

File: THCS/2022_2023/VungTau_2223/Bai2/GIFT.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    ifstream infile("GIFT.INP"); // Mở file đầu vào
    ofstream outfile("GIFT.OUT"); // Mở file đầu ra

    int N, K;
    infile >> N >> K; // Đọc số lượng phần quà và K
    vector<int> gifts(N);

    for (int i = 0; i < N; i++) {
        infile >> gifts[i]; // Đọc giá trị các phần quà
    }

    // Sắp xếp danh sách phần quà theo giá trị tăng dần
    sort(gifts.begin(), gifts.end());

    int count = 0;  // Số lượng phần quà đã chọn
    int lastGift = -1;  // Giá trị của phần quà cuối cùng đã chọn

    for (int i = 0; i < N; i++) {
        // Kiểm tra xem phần quà có thể được chọn hay không
        if (lastGift == -1 || gifts[i] >= lastGift + K) {
            count++;
            lastGift = gifts[i];  // Cập nhật giá trị của phần quà đã chọn
        }
    }

    outfile << count << endl;  // Ghi số lượng phần quà đã chọn vào file đầu ra

    infile.close();  // Đóng file đầu vào
    outfile.close(); // Đóng file đầu ra

    return 0;
}
